### DISCUSSION TITLE
Requests for monthly unlocked airstreams

### DISCUSSION THREAD

**Discussion by razgraf** [sablier-labs/v2-periphery#312] [General]
While implementing support for the "monthly" version of airstreams with MLLT in the app (actually after I almost finished it 🤦‍♂️), I released it makes no sense as you cannot predict the length of a month if you don't know the start date of each stream. While "stepper" works, as it doesn't involved fixed calendar dates, "monthly" is much more sensitive to a known start date.

E.g. If I calculate that starting today April 4th, the distance to May 4th is 30 days, if we have a user claiming on May 1st, those 30 days will bring him into May 31st, not June 1st. 

This (h/t @PaulRBerg) would probably be possible only with range(d) airstreams (#279) as that grants us access to this necessary predetermined start date for every user claiming their streams.

**Comment by PaulRBerg** [comment]
End-to-end testing can reveal issues like this sometimes 😔

I speculate that some users will be OK with stepper-like distributions, i.e. 30-day chunks, even if this will mean that the unlock events do not occur on the 1st date of each month.

But yeah, you're right that this cannot be implemented with the current design of the Merkle contracts. We need ranged airstreams as per the discussion in #279.
